app/services/course/material/download_service.rb

```diff
@@ -0,0 +1,64 @@
+class Course::Material::DownloadService
+  require 'zip'
+  require 'uri'
+  require 'open-uri'
+
+  class << self
+    # Download the contents and zip them.
+    #
+    # @param [Array<Course::Material|Course::Material::Folder>] contents The contents to be
+    #   downloaded.
+    # @return [String] The url to the zip file.
+    def download_and_zip(contents)
```

Comment:
Sure. Just think that other ppl might need to think for a while when see these two different types of parameters.